platform/mellanox/hw-management/0001-Make-hw-mgmt-SimX-compatiable.patch

```diff
@@ -1,53 +1,53 @@
-From 051938b7c49cc18aaddd699939353f591554d635 Mon Sep 17 00:00:00 2001
-From: Mykola Faryma <mykolaf@mellanox.com>
-Date: Wed, 3 Apr 2019 14:09:26 +0000
-Subject: [PATCH] Make hw-mgmt SimX compatiable.
+From c6ee8c86c35f8b1e60bf4df0d7198f349f8552c1 Mon Sep 17 00:00:00 2001
+From: Stephen Sun <stephens@mellanox.com>
+Date: Wed, 25 Dec 2019 19:33:17 +0800
+Subject: [PATCH] [PATCH] Make hw-mgmt SimX compatiable.
 
-Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>
+Signed-off-by: Stephen Sun <stephens@mellanox.com>
 ---
  usr/usr/bin/hw-management.sh | 29 +++++++++++++++++++++++++++++
  1 file changed, 29 insertions(+)
 
 diff --git a/usr/usr/bin/hw-management.sh b/usr/usr/bin/hw-management.sh
-index fdb3013..68da9bc 100755
+index cff10fe..0511c7c 100755
 --- a/usr/usr/bin/hw-management.sh
 +++ b/usr/usr/bin/hw-management.sh
-@@ -646,6 +646,35 @@ do_chip_down()
+@@ -737,6 +737,35 @@ do_chip_down()
  	/usr/bin/hw-management-thermal-events.sh change hotplug_asic down %S %p
  }
  
 +handle_simx()
 +{
-+	local -r onie_platform="$(cat /host/machine.conf | grep onie_platform | cut -d= -f2)"
++        local -r onie_platform="$(cat /host/machine.conf | grep onie_platform | cut -d= -f2)"
```

Comment:
looks like there are lots of changes but it is in the tab size. it is hard to review only the relevant changes. can you please have a PR with the real fixes only?

Reply:
we have a patch in sonic on hw-mgmt from several months before. during this time even though the hw-mgnt has been updated for several versions the patch is still able to be applied. this is because the hw-mgmt code around the patch hasn't been changed across the versions. (/usr/bin/hw-management.sh do_chip_down)
however, from the V.7.0000.2308 there is some code change in that part, which fails the patch from being applied.
So what I did is just to fix the conflicts and re-generate the patch file.